### PR TITLE
feat(breadcrumbs): Use new timeline interface for replay frames

### PR DESCRIPTION
**Pull request by leeandher**
This PR introduces a new interface for timeline items that will better match the breadcrumbs on issue details pages. It relates to [this project](https://github.com/getsentry/sentry/issues/72504).

Searching and filtering still work as is so no need to adjust any of that code, this is just an updated UI.
image
image

**todo**
- [x] Add tests 
- [x] Figure out the flickering with scrolling the virtualized list

### CODE REVIEW

**Comment by ryan953** [comment]
There's an issue where the scrollable area isn't resizing properly when there are few items in the list, causing this item to be cutoff:

SCR-20240621-iqik

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 1]
```diff
@@ -63,6 +66,91 @@ function BreadcrumbItem({
...
+
+  const renderIssueLink = useCallback(() => {
+    return isErrorFrame(frame) || isFeedbackFrame(frame) ? (
+      <CrumbErrorIssue frame={frame} />
+    ) : null;
+  }, [frame]);
+
+  const hasNewTimelineUI = useHasNewTimelineUI();
+  const timeString = new Date(frame.timestampMs).toISOString();
+  const startTimeString = new Date(startTimestampMs).toISOString();
```
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:126]: there might be `frame.timestamp` already that is a Date type

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 2]
```diff
@@ -63,6 +66,91 @@ function BreadcrumbItem({
...
+        />
+      </InspectorWrapper>
+    );
+  }, [description, expandPaths, onInspectorExpanded]);
+
+  const renderComparisonButton = useCallback(() => {
+    return frame?.data && 'mutations' in frame.data ? (
+      <div>
+        <OpenReplayComparisonButton
+          replay={replay}
```
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:94]: FYI one of us will have a minor conflict here

This is the other PR https://github.com/getsentry/sentry/pull/73095/files#diff-82c19ad3589826b66a3468b8cc00fffb9202b6b5936f7910c20e9480be9b21ed

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx (hunk 1 above)
Comment (leeandher) [reply static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:126]: It looks like `timestamp` doesn't exist on `SpanFrame` so but `timestampMs` exists on all the types of `ReplayFrame`s so I might have to leave it as is.

**Comment by ryan953** [comment]
Crumbs in the timeline look good, font is a bit small/thin

SCR-20240621-jfvm

**Review by michellewzhang — COMMENTED** [review]
SCR-20240621-kgsl
 
mobile replay breadcrumbs are looking good!

**Review by michellewzhang — APPROVED** [review]

**Comment by leeandher** [comment]
> There's an issue where the scrollable area isn't resizing properly when there are few items in the list, causing this item to be cutoff:

I wasn't able to reproduce this with a [replay](https://sentry.dev.getsentry.net:7999/replays/7b7771f9c719412987b1f534ce254810/?newTimeline=1&project=11276&query=&referrer=%2Fissues%2F%3AgroupId%2Freplays%2F&statsPeriod=1h&t_main=breadcrumbs&yAxis=count()) with only a few frames:
image

I'm going to merge this as is but I'll be revisiting the design a bit more so if it does appear to be a consistent issue, please add [it here](https://github.com/getsentry/sentry/issues/72504), and I'll take care of it 🚢

**Pull request merged**